examples/mmap_lazyio.jl

```diff
@@ -1,58 +1,78 @@
-# # Memory-mapping TIFFs
+# # Lazy TIFFs
 
 # If you're running into memory-limitations when working with large datasets,
-# you can memory-map the file so that it looks and behaves as if it were loaded,
-# but it instead lazily loads data only when needed.
+# you can lazy-load or memory-map the file so that it looks and behaves as if it were loaded,
+# but it instead loads data only when needed.
 
 #md # ```@contents
-#md # Pages = ["mmap.md"]
+#md # Pages = ["mmap_lazyio.md"]
 #md # Depth = 5
 #md # ```
 
 get_example(name) = download("https://github.com/tlnagy/exampletiffs/blob/master/$name?raw=true") #hide
 filepath = get_example("mri.tif");                                                                #hide
 
-# ### Memory-mapped loading
+# ### Memory-mapping and lazy loading
 
 # Loading is very similar to [Reading TIFFs](@ref), except with the addition of
-# the `mmap=true` flag. Here, we're loading `mri.tif` from the
+# the `mmap=true` or `lazyio=true` flag. Here, we're loading `mri.tif` from the
 # [`tlnagy/exampletiffs`](https://github.com/tlnagy/exampletiffs) repo
 
+# `mmap=true` yields good indexing performance in a wider array of circumstances
+# than `lazyio=true`, but `mmap` cannot handle compressed files or ones where data
+# are stored in [strips](https://en.wikipedia.org/wiki/TIFF#Overview).
+# Since this is a compressed TIFF, here we use `lazyio`:
+
 using TiffImages
-img = TiffImages.load(filepath; mmap=true);
+img = TiffImages.load(filepath; lazyio=true);
 
-# The memory-mapped img will behave much the same as a normal eagerly loaded
+# The lazily-loaded img will behave much the same as a normal eagerly loaded
 # image:
 
 size(img)
 
 # Display the 2nd slice
 
-img[:, :, 2] 
+img[:, :, 2]
 
-# Currently, TiffImages does not support inplace mutating operations on
-# memory-mapped files. That is, using `setindex!` and friends will throw an
-# error.
+# Currently, TiffImages does not support inplace mutating operations on `lazyio`
+# images: that is, using `setindex!` and friends will throw an error.
 
 #img[:, :, 2] .= 0.0 # throws an error
 
+# However, for files that can be supported by `mmap=true`, you can set values
+# as long as you load the image with
+
+#img = TiffImages.load(filepath; mmap=true, mode="r+")
```

Comment:
Otherwise, this will a plain markdown line with no syntax highlight.

img = TiffImages.load(filepath; mmap=true, mode="r+")

````suggestion
# ```julia
# img = TiffImages.load(filepath; mmap=true, mode="r+")
# ```
````